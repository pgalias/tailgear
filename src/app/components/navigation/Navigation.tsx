import React, { FunctionComponent } from 'react';
import styles from './navigation.module.css';
import { useSections } from '../../store/components';
import { Section, StaticPage } from './components';
import { StaticPages } from '../../constants/staticPages';

export const Navigation: FunctionComponent = () => {
  const sections = useSections();

  return (
    <nav className={styles.navigation}>
      <div className={styles.wrapper}>
        {StaticPages.map((page) => (
          <StaticPage page={page} key={page.title} />
        ))}
        {sections.map((section) => (
          <Section section={section} key={section.id} />
        ))}
      </div>
    </nav>
  );
};
