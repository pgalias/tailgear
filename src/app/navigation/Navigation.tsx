import React, { FunctionComponent } from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome, faCodeBranch } from '@fortawesome/free-solid-svg-icons';
import { ComponentsList } from '../components';
import styles from './navigation.module.css';
import { changeMode, useDispatch } from '../store/layout';

// TODO: Make ComponentList generic
// let's take a component list for that and which mode should represent
export const Navigation: FunctionComponent = () => {
  const dispatch = useDispatch();
  const setStaticMode = () => dispatch(changeMode('static'));

  return (
    <nav className={styles.navigation}>
      <div className={styles.wrapper}>
        <div className={styles.section}>
          <h3>
            <NavLink
              exact
              to="/"
              activeClassName={styles.currentLink}
              onClick={setStaticMode}
            >
              <FontAwesomeIcon icon={faHome} />
              <span> Home</span>
            </NavLink>
          </h3>
        </div>
        <div className={styles.section}>
          <h3>
            <NavLink
              exact
              to="/contribution"
              activeClassName={styles.currentLink}
              onClick={setStaticMode}
            >
              <FontAwesomeIcon icon={faCodeBranch} />
              <span> Contribution</span>
            </NavLink>
          </h3>
        </div>
        <ComponentsList
          activeClassName={styles.currentLink}
          sectionClassName={styles.section}
          blockClassName={styles.block}
        />
      </div>
    </nav>
  );
};
