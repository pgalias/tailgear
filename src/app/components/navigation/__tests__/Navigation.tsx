import React from 'react';
import { render, screen } from '@testing-library/react';
import { Navigation } from '../Navigation';

describe('app::components::Navigation', () => {
  test('should render properly', () => {
    render(
      <Navigation>
        <ul>
          <li>
            <a href="/foo">foo</a>
          </li>
          <li>
            <a href="/bar">bar</a>
          </li>
        </ul>
        <button type="button">btn</button>
      </Navigation>
    );

    expect(screen.getByRole('navigation')).toMatchSnapshot();
  });
});
