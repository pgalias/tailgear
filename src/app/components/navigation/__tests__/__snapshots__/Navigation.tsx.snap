// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app::components::Navigation should render properly 1`] = `
<nav
  class="navigation"
>
  <div
    class="wrapper"
  >
    <ul>
      <li>
        <a
          href="/foo"
        >
          foo
        </a>
      </li>
      <li>
        <a
          href="/bar"
        >
          bar
        </a>
      </li>
    </ul>
    <button
      type="button"
    >
      btn
    </button>
  </div>
</nav>
`;
